common/deftype/deftype.cpp

```diff
@@ -1427,9 +1435,29 @@ bool CModifierTypeInfo::equals(const CTypeInfo & _other) const
     return (baseType == other.baseType) && (kind == other.kind) && (extra == other.extra);
 }
 
-
 //===========================================================================
 
+template<typename T>
+T* getOrCreateSingletonAtomic(std::atomic<T*>& singleton)
```

Comment:
naming: getLinkedSingleton() would be a better name.  A singleton by definition is either retrieved or created, so no need for orCreate in the function name.  Atomic is also really redundant.